STUDIO/org.openl.rules.repository.git/src/org/openl/rules/repository/git/GitRepository.java

```diff
@@ -352,28 +356,24 @@ private boolean deleteInternal(FileData data) throws IOException {
 
                 String markerFile = name + "/" + DELETED_MARKER_FILE;
                 git.add().addFilepattern(markerFile).call();
-                RevCommit commit = git.commit()
+                CommitCommand commitCommand = git.commit()
                     .setMessage(commitMessage)
-                    .setCommitter(userDisplayName != null ? userDisplayName : data.getAuthor(),
-                        userEmail != null ? userEmail : "")
                     .setOnly(markerFile)
-                    .setNoVerify(noVerify)
-                    .call();
+                    .setNoVerify(noVerify);
+                RevCommit commit = addUserInfoToCommit(commitCommand, data.getAuthor()).call();

```

Comment:
what if, there will be installed a hook that will block all commits if email and author are not related to the real committer?

Reply:
If user is external, than external display name and email will be used. If AD user trying to push, his AD email and display name will be used.